lib/floe/workflow/choice_rule.rb

```diff
@@ -4,31 +4,37 @@ module Floe
   class Workflow
     class ChoiceRule
       class << self
-        def build(payload)
+        # @param [Workflow] workflow - passed to the top level build only
+        def build(workflow, name, payload)
           if (sub_payloads = payload["Not"])
-            Floe::Workflow::ChoiceRule::Not.new(payload, build_children([sub_payloads]))
+            Floe::Workflow::ChoiceRule::Not.new(workflow, name, payload, build_children(name, [sub_payloads]))
           elsif (sub_payloads = payload["And"])
-            Floe::Workflow::ChoiceRule::And.new(payload, build_children(sub_payloads))
+            Floe::Workflow::ChoiceRule::And.new(workflow, name, payload, build_children(name, sub_payloads))
           elsif (sub_payloads = payload["Or"])
-            Floe::Workflow::ChoiceRule::Or.new(payload, build_children(sub_payloads))
+            Floe::Workflow::ChoiceRule::Or.new(workflow, name, payload, build_children(name, sub_payloads))
           else
-            Floe::Workflow::ChoiceRule::Data.new(payload)
+            Floe::Workflow::ChoiceRule::Data.new(workflow, name, payload)
           end
         end
 
-        def build_children(sub_payloads)
-          sub_payloads.map { |payload| build(payload) }
+        def build_children(name, sub_payloads)
+          # passing workflow = nil because it is not at the top level
+          sub_payloads.map { |payload| build(nil, name, payload) }
```

Comment:
🍅 Not sure that I like not passing `workflow` meaning it is a lower level. (I don't like passing name/workflow into here at all, but I digress)

Do you like `:top => false` instead?